src/js/_enqueues/lib/emoji-loader.js

```diff
@@ -164,12 +227,33 @@
 		everythingExceptFlag: true
 	};
 
+	// Initialize sessionSupports from sessionStorage if available. This avoids expensive calls to browserSupportsEmoji().
+	if (
+		typeof sessionStorage !== "undefined" &&
+		sessionStorageKey in sessionStorage
+	) {
+		try {
+			sessionSupports = JSON.parse(sessionStorage.getItem(sessionStorageKey));
+			Object.assign(settings.supports, sessionSupports);
+		} catch (e) {
+			sessionSupports = {};
+		}
+	} else {
+		sessionSupports = {};
+	}
+
 	/*
 	 * Tests the browser support for flag emojis and other emojis, and adjusts the
 	 * support settings accordingly.
 	 */
 	for( ii = 0; ii < tests.length; ii++ ) {
-		settings.supports[ tests[ ii ] ] = browserSupportsEmoji( tests[ ii ] );
+		if (!(tests[ii] in sessionSupports)) {
+			sessionSupports[tests[ii]] = await browserSupportsEmojiOptimized(
+				tests[ii]
+			);
+			settings.supports[tests[ii]] = sessionSupports[tests[ii]];
```

Comment:
At first glance that seems to be correct, but it seems we need to keep looping over everything because we need to set `everythingExceptFlag` properly. At second glance, this doesn't also seem to be needed, but I did a quick `ack` and found it is used in `emoji.js`:

https://github.com/WordPress/wordpress-develop/blob/ba9a2f8b83211fbdee1621fd4f39dcb11908b817/src/js/_enqueues/wp/emoji.js#L236

So we still do need to loop over everything, I believe.